### PR TITLE
DPE-2154 profile configuration support

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

No profile support

## Solution

Profile support as per [spec](https://docs.google.com/document/d/1bq7KyUAPIIcUWGgifW05p40v7lGXs3PhMznM2ECaMKY/edit)

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -5,6 +5,13 @@ options:
   cluster-name:
     description: "Optional - Name of the MySQL InnoDB cluster"
     type: "string"
+  profile:
+   description: |
+      profile representing the scope of deployment, and used to be able to enable high-level
+      high-level customisation of sysconfigs, resource checks/allocation, warning levels, etc.
+      Allowed values are: “production” and “testing”.
+   type: string
+   default: production
```
Comment (taurus-forever) [inline config.yaml:14]: CC: @delgod , CC: mohamed, please approve the option name.

**Review by taurus-forever — APPROVED** [review]
LGTM, requested Mohamed ACK for the option name.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -108,7 +108,6 @@ jobs:
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: lxd
-          bootstrap-options: "--agent-version 2.9.42"  # renovate: latest
```
[the remark this answers is not in the document]
Comment (paulomach) [reply .github/workflows/ci.yaml:111]: To use stable/latest. I see no point in pinning version here

File: config.yaml (hunk 1 above)
Comment (paulomach) [reply config.yaml:14]: Sure. This `yyp` from the spec doc though

File: src/charm.py [hunk 3]
```diff
@@ -578,7 +578,7 @@ def _workload_initialise(self) -> None:
         Create users and configuration to setup instance as an Group Replication node.
         Raised errors must be treated on handlers.
         """
-        self._mysql.create_custom_mysqld_config()
+        self._mysql.create_custom_mysqld_config(profile=self.config.get("profile"))
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/charm.py]: dfcbde3

File: .github/workflows/ci.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulomach) [reply .github/workflows/ci.yaml:111]: fair point

**Review by shayancanonical — APPROVED** [review]

**Pull request merged**